drivers/gpio/gpio_raspberrypi.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2021, Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <drivers/gpio.h>
+#include <hardware/gpio.h>
+#include <hardware/regs/intctrl.h>
+#include <hardware/structs/iobank0.h>
+
+#include "gpio_utils.h"
+
+#define DT_DRV_COMPAT raspberrypi_rp2_gpio
+
+#define IRQ_PRIORITY 0
+#define ALL_EVENTS (GPIO_IRQ_EDGE_FALL | GPIO_IRQ_EDGE_RISE \
+		| GPIO_IRQ_LEVEL_LOW | GPIO_IRQ_LEVEL_HIGH)
+
+struct gpio_raspberrypi_config {
+	struct gpio_driver_config common;
+};
+
+struct gpio_raspberrypi_data {
+	struct gpio_driver_data common;
+	sys_slist_t callbacks;
+	uint32_t int_enabled_mask;
+};
+
+static int gpio_raspberrypi_configure(const struct device *dev,
+				gpio_pin_t pin,
+				gpio_flags_t flags)
+{
+	if (flags & GPIO_SINGLE_ENDED) {
+		return -ENOTSUP;
+	}
+
+	gpio_init(pin);
+
+	if (flags & GPIO_OUTPUT) {
+		gpio_set_dir(pin, GPIO_OUT);
+
+		if (flags & GPIO_OUTPUT_INIT_HIGH)
+			gpio_put(pin, 1);
+		else if (flags & GPIO_OUTPUT_INIT_LOW)
+			gpio_put(pin, 0);
```

Comment:
According to the Zephyr coding style you have to use `{ }` also for on-line `if`.